File: Toll-Calculator/Models/Vehicles/Motorbike.cs
using Toll_Calculator.Enums;
using Toll_Calculator.Interfaces;

namespace Toll_Calculator.Models.Vehicles
{
    public class Motorbike : IVehicle
    {
        public bool IsTollFree()
        {
            return true;
        }

        public VehicleType GetVehicleType()
        {
            return VehicleType.Motorbike;
        }
    }
}

File: Golang/vehicle/motorbike.go
package vehicle

// motorbike struct for handling Motorbike type.
type motorbike struct{}

// GetVehicleType will return the "Motorbike" type.
func (m motorbike) GetVehicleType() string {
	return "Motorbike"
}

// IsTollFree return true for Motorbike type.
func (m motorbike) IsTollFree() bool {
	return true
}

// NewMotorbike return a vehicle type of Motorbike.
func NewMotorbike() Vehicle {
	return &motorbike{}
}
